### PR TITLE
[url_launcher] Documentation improvement for file scheme to readme

### PR DESCRIPTION
**Pull request by jpsiitonen**
Update for documentation on readme file:
- describes file:/ scheme usage, related to platforms: macOS, linux and Windows.
- link update to `Uri` class documentation

Fixes https://github.com/flutter/flutter/issues/97422

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/url_launcher/url_launcher/README.md [hunk 1]
```diff
@@ -96,16 +96,18 @@ takes a string argument containing a URL. This URL
 can be formatted using a number of different URL schemes. The supported
 URL schemes depend on the underlying platform and installed apps.
 
-Common schemes supported by both iOS and Android:
+Common schemes supported:
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/README.md:99]: Let's changed this to "Commonly used schemes include:"

"Supported" sends the wrong message here (which is addressed in the sentence above, but we shouldn't muddy the waters here).

File: packages/url_launcher/url_launcher/README.md [hunk 2]
```diff
@@ -96,16 +96,18 @@ takes a string argument containing a URL. This URL
...
-| `http:<URL>` , `https:<URL>`, e.g. `http://flutter.dev` | Open URL in the default browser |
-| `mailto:<email address>?subject=<subject>&body=<body>`, e.g. `mailto:smith@example.org?subject=News&body=New%20plugin` | Create email to <email address> in the default email app |
-| `tel:<phone number>`, e.g. `tel:+1 555 010 999` | Make a phone call to <phone number> using the default phone app |
-| `sms:<phone number>`, e.g. `sms:5550101234` | Send an SMS message to <phone number> using the default messaging app |
+|:---|:---|
+| `http:<URL>` , `https:<URL>`, e.g `http://flutter.dev` | Open URL in the default browser |
+| `mailto:<email address>?subject=<subject>&body=<body>`, e.g `mailto:smith@example.org?subject=News&body=New%20plugin` | Create email to <email address> in the default email app |
+| `tel:<phone number>`, e.g `tel:+1 555 010 999` | Make a phone call to <phone number> using the default phone app |
+| `sms:<phone number>`, e.g `sms:5550101234` | Send an SMS message to <phone number> using the default messaging app |
+| `file:/<path>`, e.g `file:/home` | Opens file or folder using default app association, supported on desktop platforms |
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/README.md:107]: `file:<path>`. `/` is part of the path.

File: packages/url_launcher/url_launcher/README.md [hunk 3]
```diff
@@ -164,3 +166,39 @@ If you do this for a URL of a page containing JavaScript, make sure to pass in
 `enableJavaScript: true`, or else the launch method will not work properly. On
 iOS, the default behavior is to open all web URLs within the app. Everything
 else is redirected to the app handler.
+
+## File scheme handling
+`file:` scheme can be used on desktop platforms: `macOS`, `Linux` and `Windows`.
+
+We recommend checking first whether the directory and/or file exists before calling the `launch`.
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/README.md:173]: Something can't be both a directory and a file, so "and/or" should just be "or".

File: packages/url_launcher/url_launcher/README.md [hunk 4]
```diff
@@ -96,16 +96,18 @@ takes a string argument containing a URL. This URL
...
+Common schemes supported:
 
 | Scheme | Action |
-|---|---|
-| `http:<URL>` , `https:<URL>`, e.g. `http://flutter.dev` | Open URL in the default browser |
-| `mailto:<email address>?subject=<subject>&body=<body>`, e.g. `mailto:smith@example.org?subject=News&body=New%20plugin` | Create email to <email address> in the default email app |
-| `tel:<phone number>`, e.g. `tel:+1 555 010 999` | Make a phone call to <phone number> using the default phone app |
-| `sms:<phone number>`, e.g. `sms:5550101234` | Send an SMS message to <phone number> using the default messaging app |
+|:---|:---|
+| `http:<URL>` , `https:<URL>`, e.g `http://flutter.dev` | Open URL in the default browser |
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/README.md:103]: Why did you change "e.g." to be incorrect everywhere in this table?

Comment (stuartmorgan) [reply packages/url_launcher/url_launcher/README.md:103]: I think we could actually make this table easier to read by changing it to:

```
| Scheme | Example | Action |
| `http:<URL>` , `https:<URL>` | `http://flutter.dev` | Open URL in the default browser |
...
```

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/url_launcher/url_launcher/README.md [hunk 5]
```diff
@@ -96,15 +96,15 @@ takes a string argument containing a URL. This URL
...
-| `http:<URL>` , `https:<URL>`, e.g `http://flutter.dev` | Open URL in the default browser |
-| `mailto:<email address>?subject=<subject>&body=<body>`, e.g `mailto:smith@example.org?subject=News&body=New%20plugin` | Create email to <email address> in the default email app |
-| `tel:<phone number>`, e.g `tel:+1 555 010 999` | Make a phone call to <phone number> using the default phone app |
-| `sms:<phone number>`, e.g `sms:5550101234` | Send an SMS message to <phone number> using the default messaging app |
-| `file:/<path>`, e.g `file:/home` | Opens file or folder using default app association, supported on desktop platforms |
+| Scheme | Example | Action |
+|:---|:---|:---|
+| `https:<URL>` | `https://flutter.dev` | Open URL in the default browser |
+| `mailto:<email address>?subject=<subject>&body=<body>` | `mailto:smith@example.org?subject=News&body=New%20plugin` | Create email to <email address> in the default email app |
+| `tel:<phone number>` | `tel:+1 555 010 999` | Make a phone call to <phone number> using the default phone app |
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/README.md:105]: Sorry, one more nit since you are touching these lines: I just noticed that this example is actively wrong (this isn't a valid URL because of the spaces). We should fix it while we are here to `tel:+1-555-010-999`.

File: packages/url_launcher/url_launcher/README.md [hunk 6]
```diff
@@ -186,19 +186,5 @@ if (await File(uri.toFilePath(windows: Platform.isWindows)).exists()) {
...
-```xml
-<dict>
-    <key>com.apple.security.app-sandbox</key>
-    <true/>
-    <key>com.apple.security.files.user-selected.read-only</key>
-    <true/>
-</dict>
-```
+If you need to access files outside of your application's sandbox, you will need to have the necessary 
+[entitlements](https://docs.flutter.dev/desktop#entitlements-and-the-app-sandbox)
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/README.md:190]: Nit: There's a missing period at the end of this sentence.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
@bparrishMines for a quick secondary review.

File: packages/url_launcher/url_launcher/README.md (hunk 5 above)
Comment (jpsiitonen) [reply packages/url_launcher/url_launcher/README.md:105]: No, problem. Updated tel just like you suggested.

File: packages/url_launcher/url_launcher/README.md [hunk 7]
```diff
@@ -96,16 +96,18 @@ takes a string argument containing a URL. This URL
...
-| `mailto:<email address>?subject=<subject>&body=<body>`, e.g. `mailto:smith@example.org?subject=News&body=New%20plugin` | Create email to <email address> in the default email app |
-| `tel:<phone number>`, e.g. `tel:+1 555 010 999` | Make a phone call to <phone number> using the default phone app |
-| `sms:<phone number>`, e.g. `sms:5550101234` | Send an SMS message to <phone number> using the default messaging app |
+| Scheme | Example | Action |
+|:---|:---|:---|
+| `https:<URL>` | `https://flutter.dev` | Open URL in the default browser |
+| `mailto:<email address>?subject=<subject>&body=<body>` | `mailto:smith@example.org?subject=News&body=New%20plugin` | Create email to <email address> in the default email app |
+| `tel:<phone number>` | `tel:+1-555-010-999` | Make a phone call to <phone number> using the default phone app |
+| `sms:<phone number>` | `sms:5550101234` | Send an SMS message to <phone number> using the default messaging app |
+| `file:<path>` | `file:/home` | Opens file or folder using default app association, supported on desktop platforms |
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/README.md:107]: Sorry, one more nit 😐 : This should be "Open" instead of "Opens" to match all of the others.

File: packages/url_launcher/url_launcher/README.md [hunk 8]
```diff
@@ -164,3 +166,25 @@ If you do this for a URL of a page containing JavaScript, make sure to pass in
...
+
+Example:
+```dart
+var filePath = '/path/to/file';
+final Uri uri = Uri.file(filePath);
+
+if (await File(uri.toFilePath()).exists()) {
+    if (!await launch(uri.toString())) {
+      throw 'Could not launch $uri';
+    }
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/README.md:183]: And one more: this body is indented 4 instead of 2, making it inconsistent with the inner if's indentation of 2 (and normal Dart practice of indenting 2)

**Review by stuartmorgan — APPROVED** [review]
LGTM with two last nits.

**Review by bparrishMines — APPROVED** [review]
LGTM
